File: NetFramework/XHtmlEditor/CS/RichTextEditor/AppMenuTabs/SaveAsTab.cs
using RichTextEditor.AppMenuTabs.Items;
using System;
using System.Drawing;

namespace RichTextEditor.AppMenuTabs
{
    public partial class SaveAsTab : AppMenuTabFileList
    {
        public SaveAsTab()
        {
            Button.ItemImage = Properties.Resources.Open_large;
            Button.ItemText = Properties.Resources.Browse_txt;
            Caption = GetElementText(Properties.Resources.SaveAsTab_txt);
        }

        #region override
        protected override void OnMenuItemClick(AppMenuTabItem mi)
        {
            var ribbon = (C1TextEditorRibbon)RibbonApplicationMenu.Ribbon;
            ribbon.SaveDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text));
            mi.Item.Date = DateTime.Now;
        }

        protected override void OnMenuButtonClick()
        {
            var ribbon = (C1TextEditorRibbon)RibbonApplicationMenu.Ribbon;
            ribbon.SaveDocumentAs();
        }
        #endregion  
    }
}

File: NetFramework/XHtmlEditor/VB/RichTextEditor/AppMenuTabs/SaveAsTab.vb

Imports RichTextEditor.RichTextEditor

Partial Public Class SaveAsTab
    Inherits AppMenuTabFileList

    Public Sub New()
        Button.ItemImage = Resources.Open_large
        Button.ItemText = Resources.Browse_txt
        Caption = GetElementText(Resources.SaveAsTab_txt)
    End Sub

    Protected Overrides Sub OnMenuItemClick(ByVal mi As AppMenuTabItem)
        Dim ribbon As C1TextEditorRibbon = CType(RibbonApplicationMenu.Ribbon, C1TextEditorRibbon)
        ribbon.SaveDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text))
        mi.Item.ItemDate = DateTime.Now
    End Sub

    Protected Overrides Sub OnMenuButtonClick()
        Dim ribbon As C1TextEditorRibbon = CType(RibbonApplicationMenu.Ribbon, C1TextEditorRibbon)
        ribbon.SaveDocumentAs()
    End Sub
End Class
